keylime/da/record.py

```diff
@@ -0,0 +1,337 @@
+                agent_data["agent_id"],
+            )
+
+    def _record_timestamp_check(self, record_object: dict, record_identifier: str, contents: bytes):
+        """Accesses a Time Stamp Autorhity (TSA) and checks a signed timestamp for a given contents"""
+        if self.tsa_url.scheme and contents:
+
+            logger.debug(
+                "Checking the time stamp for data referring to agent %s from TSA at %s",
+                record_identifier,
+                self.tsa_url._replace(fragment="").geturl(),
+            )
+
+            getattr(importlib.import_module(self.st_imp_path + ".tsa_rfc3161"), "record_timestamp_check")(
+                record_object, record_identifier, contents, self.tsa_url, self.tsa_cert
+            )
+
+    def _record_signature_check(self, record_object: dict, agent_data: dict) -> str:
+        """Check the signature for contents of a given record"""
+
+        if not "signature" in record_object:
+            return None
+
+        self._get_certs_path()
+
+        with tempfile.TemporaryDirectory() as _temp_dir_path:
+
+            record_object["temp_dir"] = _temp_dir_path
+            record_object["contents_file_path"] = f'{record_object["temp_dir"]}/object.json'
+            record_object["signature_file_path"] = f'{record_object["temp_dir"]}/object.json.sig'
+
+            with open(record_object["signature_file_path"], "wb") as fp:
+                try:
+                    record_object["signature"] = str.encode(record_object["signature"])
+                except Exception:
+                    pass
+                fp.write(base64.decodebytes(record_object["signature"]))
+
+            with open(self.key_tls_pub, "w", encoding="utf-8") as fp:
+                fp.write(record_object["signer_pub_key"])
+
+            _contents_to_check = {}
+            _contents_to_check["agent"] = record_object["agent"]
+
+            _contents_to_check = json.dumps(_contents_to_check).encode("utf-8")
+
+            with open(record_object["contents_file_path"], "wb") as fp:
+                fp.write(_contents_to_check)
+
+            if "contents_file_path" in record_object and "signature_file_path" in record_object:
+                if self.tl_url.scheme == "http" and self.tl_url.netloc.count("3000"):
+                    getattr(importlib.import_module(self.st_imp_path + ".rekor"), "record_signature_check")(
+                        record_object, agent_data, self.tl_url, self.key_tls_pub
+                    )
+
+        return _contents_to_check
+
+
+class RecordManagementException(Exception):
+    pass
+
+
+def build_key_list(registration_record_list: list, aik_list: list):
+    """Just assembles a simple list of AIKs used by an agent"""
+
+    for _entry in registration_record_list:
+        if _entry["agent"]["aik_tpm"] not in aik_list:
+            logger.debug("New AIK added to the list")
+            aik_list.append(_entry["agent"]["aik_tpm"])
+
+
+def mkdir_p(path: str):
```

Comment:
Drop this, and use `os.makedirs(name, exist_ok=True)`

Reply:
y